cmd/subnetcmd/upgradecmd/vm.go

```diff
@@ -280,8 +290,51 @@ func updateFutureVM(sc models.Sidecar, targetVersion string) error {
 	return nil
 }
 
-func updateExistingLocalVM(_ models.Sidecar, _ string) error {
-	ux.Logger.PrintToUser("Coming soon. For now, please upgrade your existing deployments and redeploy the subnet.")
+func updateExistingLocalVM(sc models.Sidecar, targetVersion string) error {
+	// check network has been stopped
+	cli, err := binutils.NewGRPCClient()
+	if err != nil {
+		return err
+	}
+	ctx := binutils.GetAsyncContext()
+
+	_, err = cli.Status(ctx)
+
+	if err == nil || !server.IsServerError(err, server.ErrNotBootstrapped) {
+		ux.Logger.PrintToUser("Please stop network before upgrading local VMs")
+	}
+
+	vmid, err := utils.VMID(sc.Name)
+	if err != nil {
+		return err
+	}
+	var vmBin string
+	switch sc.VM {
+	// setup the binary for the new VM
+	case models.SubnetEvm:
+		vmBin, err = binutils.SetupSubnetEVM(app, targetVersion)
+		if err != nil {
+			return fmt.Errorf("failed to install subnet-evm: %w", err)
+		}
+	case models.SpacesVM:
+		// update the binary in the network runner's active directory and restart the network
+		vmBin, err = binutils.SetupSpacesVM(app, targetVersion)
+		if err != nil {
+			return fmt.Errorf("failed to install spaces-vm: %w", err)
+		}
+	case models.CustomVM:
+		vmBin = binutils.SetupCustomBin(app, sc.Name)
+	default:
+		return errors.New("unknown VM type")
+	}
+
+	// Update the binary in the network runner's active directory and restart the network
```

Comment:
This code is not restarting the network

Reply:
Good catch, I didn't need to change this since I decided to skip restarting the network. I reverted this change.